File: TDD Boot Camp/Part 01/Code/Java/OOPrinciples/src/com/oop/polymorphism/Car.java
package com.oop.polymorphism;

public class Car implements Exhaustable {
	private Engine engine;
	Car(Engine engine){
		this.engine = engine;
	}
	
    public void startEngine() {
       engine.start();
    }
    public void shutDownEngine() {
    	engine.start();
    }

	@Override
	public void Exhaust() {
		//Do exhaust		
	}
}

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Polymorphism/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Polymorphism
{
    public class Car : Exhaustable
    {
        private Engine _engine { get; set; }
        public Car(Engine engine)
        {
            _engine = engine;
        }

        public void StartEngine()
        {
            _engine.Start();
        }
        public void ShutDownEngine()
        {
            _engine.Stop();
        }

        public void Exhaust()
        {
            //Do exhaust		
        }
    }
}
